### DISCUSSION TITLE
How to measure request stages?

### DISCUSSION THREAD

**Discussion by krutoo** [nodejs/undici#1313] [Q&A] [answered]
Hi, how can i measure some request stages timings like:

- DNS lookup
- time to first byte
- TCP connection established
- TLS Handshake

using `undici`? I want to listen its events for example

**Comment by mcollina** [comment, marked as answer]
I don't think this fully possible right now. We have some diagnostics_channel support but we might need more. Would you like to work on it?

**Reply by krutoo** [reply]
@mcollina I'd be happy to work on it, but I have a full-time job, so it won't be quick

**Comment by sibelius** [comment]
what progress on this ?
